mkosi

```diff
@@ -2313,14 +2314,33 @@ def detect_distribution():
 
     id = None
     version_id = None
+    version_codename = None
+    extracted_codename = None
 
     for ln in f:
         if ln.startswith("ID="):
             id = ln[3:].strip()
         if ln.startswith("VERSION_ID="):
             version_id = ln[11:].strip()
+        if ln.startswith("VERSION_CODENAME="):
+            version_codename = ln[17:].strip()
+        if ln.startswith("VERSION="):
+            # extract Debain release codename
+            version_str = ln[8:].strip()
+            debian_codename_re = r'\((.*?)\)'
+
+            codename_list = re.findall(debian_codename_re, version_str)
+            if len(codename_list) == 1:
+                extracted_codename = codename_list[0]
 
     d = Distribution.__members__.get(id, None)
+
+    if d == Distribution.debian and (version_codename or extracted_codename):
```

Comment:
perhaps a short comment after this to explain why it’s necessary? e. g. “debootstrap needs codenames, not version numbers”

Reply:
Added